serverless/networking/serverless-ossm-jwt.adoc

```diff
@@ -0,0 +1,154 @@
+
+.Procedure
+. Create a Knative service with sidecar injection, or enable sidecar injection for an existing service.
++
+[IMPORTANT]
+====
+Adding sidecar injection to Pods in system namespaces such as `knative-serving` and `knative-serving-ingress` is not supported.
+====
++
+.. Create a Knative service with sidecar injection.
+... Copy the following into a YAML file:
++
+
+[source,yaml]
+----
+  apiVersion: serving.knative.dev/v1
+  kind: Service
+  metadata:
+    name: hello-example
+  spec:
+    template:
+      metadata:
+        name: hello-example-1
+        annotations:
+          sidecar.istio.io/inject: "true"
+      spec:
+        containers:
+        - image: gcr.io/knative-samples/helloworld-go
+          name: user-container
+    traffic:
+    - latestRevision: true
+      percent: 100
+----
+
+... Apply the YAML file:
+
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+.. Enable sidecar injection to an existing Knative service by adding the `sidecar.istio.io/inject="true"` annotation to the Service resource:
++
+
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true" <1>
+    spec:
+      containers:
+      - image: docker.io/openshift/hello-openshift
+        name: container
+----
+<1> Add the `sidecar.istio.io/inject="true"` annotation.
+
+. Create a policy that only allows requests with valid JSON Web Tokens (JWT) in your serverless application namespace:
+
+.. Copy the following into a YAML file:
++
+
+[source,yaml]
+----
+apiVersion: authentication.istio.io/v1alpha1
+kind: Policy
+metadata:
+  name: default
+spec:
+  origins:
+  - jwt:
+      issuer: testing@secure.istio.io
+      jwksUri: "https://raw.githubusercontent.com/istio/istio/release-1.6/security/tools/jwt/samples/jwks.json"
+      triggerRules:
+      - excludedPaths: <1>
+        - prefix: /metrics
+        - prefix: /healthz
+  principalBinding: USE_ORIGIN
+----
+
+<1> `/metrics` and `/healthz` must be included in `excludedPaths`, as they are accessed from system pods in the `knative-serving` namespace.
+
+.. Apply the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+.Verification steps
+
+. If you try to use a `curl` request, it is denied.
++
+.Example `curl` request
++
+
+[source,terminal]
+----
+$ curl http://hello-example-default.apps.knakayam-dev9.devcluster.openshift.com/
+----
+
++
+.Example output
++
+
+[source,terminal]
+----
+Origin authentication failed.
+----
+
+. Verify the request with a valid JWT.
+.. Get the valid JWT token by entering the following command:
++
+
+[source,terminal]
+----
+$ TOKEN=$(curl https://raw.githubusercontent.com/istio/istio/release-1.6/security/tools/jwt/samples/demo.jwt -s) && echo "$TOKEN" | cut -d '.' -f2 - | base64 --decode -
+----
+
+. Access the service by using the valid token in the header.
```

Comment:
In the rendered document this sentence is numbered as `3.` but I'm wondering if this should be just `2. a.` because it's a one step in the section "Verify the request with a valid JWT"